modules/test/playwright/tests/frontend-data-set-views-web/pages/FiltersPage.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page} from '@playwright/test';
+
+import {ISelectionFilter} from '../utils/types';
+import {ViewPage} from './view/ViewPage';
+
+export class FiltersPage {
+	readonly filtersHeading: Locator;
+	readonly filtersTab: Locator;
+	readonly newFilterButton: Locator;
+	readonly newFilterTypeSelect: {
+		clientExtension: Locator;
+		dateRange: Locator;
+		selection: Locator;
+	};
+	readonly newSelectionFilterForm: {
+		filterBySelect: Locator;
+		filterModeRarioButtons: Locator;
+		nameInput: Locator;
+		newFilterModalheading: Locator;
+		picklistDropdown: Locator;
+		preselectedValuesMultiSelect: Locator;
+		saveButton: Locator;
+		selectionRadioButtons: Locator;
+		sourceDropdown: Locator;
+	};
+	readonly page: Page;
+	readonly viewPage: ViewPage;
+
+	constructor(page: Page) {
+		this.filtersHeading = page.getByRole('heading', {name: 'Filters'});
+		this.filtersTab = page.getByRole('button', {name: 'Filters'});
+		this.newFilterButton = page.getByLabel('Add').first();
+		this.newFilterTypeSelect = {
+			clientExtension: page.getByRole('menuitem', {
+				name: 'Client Extension',
+			}),
+			dateRange: page.getByRole('menuitem', {name: 'Date Range'}),
+			selection: page.getByRole('menuitem', {name: 'Selection'}),
+		};
+		this.newSelectionFilterForm = {
+			filterBySelect: page.getByRole('button', {name: 'Select'}),
+			filterModeRarioButtons: page.getByText('Filter ModeIncludeExclude'),
+			nameInput: page.getByPlaceholder('Add a name'),
+			newFilterModalheading: page.getByRole('heading', {
+				name: 'New Selection Filter',
+			}),
+			picklistDropdown: page.getByLabel('Picklist'),
+			preselectedValuesMultiSelect: page.getByPlaceholder(
+				'Select a default value for your filter.'
+			),
+			saveButton: page.getByRole('button', {name: 'Save'}),
+			selectionRadioButtons: page.getByText('SelectionMultipleSingle'),
+			sourceDropdown: page.getByLabel('Choose an Option'),
+		};
+		this.page = page;
+		this.viewPage = new ViewPage(page);
+	}
+
+	async goto({
+		dataSetLabel,
+		viewLabel,
+	}: {
+		dataSetLabel: string;
+		viewLabel: string;
+	}) {
+		await this.viewPage.goto({
+			dataSetLabel,
+			viewLabel,
+		});
+
+		await Promise.all([
+			this.viewPage.selectTab('Filters'),
+			this.page.waitForResponse(
+				(resp) =>
+					resp.status() === 200 &&
+					resp.url().includes('/openapi.json')
+			),
+		]);
+	}
```

Comment:
This is a bit unusual, waiting for rest response. A typical pattern is to wait for something in UI. If you are relying on this, it could be a source of flakiness, as UI is updated some time after the request resolves.

Reply:
`selectTab` has a [pretty robust logic to avoid flakiness](https://github.com/liferay/liferay-portal/blob/master/modules/test/playwright/tests/frontend-data-set-views-web/pages/view/ViewPage.ts#L37-L49), so you can probably just remove this.

Reply:
Ooh nevermind, I found out why this is needed. Not doing this was the cause of my flakiness issues in https://github.com/liferay-frontend/liferay-portal/pull/4157

Nice job here :+1:

Reply:
It's not a blocker for this PR. We can merge `waitForResponse`s, and remove then along with the bugfix.